File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        curr_max=0
        l=0
        h=len(height)-1
        while l<h:
            curr_max=max(curr_max,(min(height[l],height[h])*(h-l)))
            if height[l]<height[h]:
                l=l+1
            else:
                h=h-1
        return curr_max
            

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0;
        int j=height.size()-1;
        long int ans=0;
        while(i<j){
            long int area=(j-i)*min(height[i],height[j]);
            ans=max(ans,area);
            if (height[i] < height[j])
                i++;
            else
                j--;
        }
        return ans;
    }
};